libs/hdf-converters/src/asff-mapper/case-hdf2asff.ts

```diff
@@ -154,6 +154,40 @@ function getCodeForProfileLayer(
   }
 }
 
+function getPassthrough(execTypes: Record<string, unknown>) {
+  //Default behavior for old instances of HDF
+  if (_.has(execTypes, 'Execution.passthrough')) {
+    return _.get(execTypes, 'Execution.passthrough');
+  }
+  //If only one field exists, then is already an object
+  if (_.has(execTypes, 'Execution.passthrough1of1')) {
+    return _.get(execTypes, 'Execution.passthrough1of1');
+  }
+  //Else reconstitute passthrough from detected strings
+  const keysArr = [];
+  const strArr = [];
+  for (const key of _.keys(execTypes.Execution)) {
+    if (/passthrough.*of/.test(key)) {
+      keysArr.push(_.pick(execTypes.Execution, key));
+    }
+  }
+  const cntMax = keysArr.length;
+  let cntMin = 1;
+  //No valid passthrough
+  if (cntMax <= 1) {
+    return {};
+  }
+  while (cntMin <= cntMax) {
```

Comment:
is there that high a concern that the array will become out of order as it goes through the systems?  I would've presumed it would've left that stuff as is, but if you experimentally found that the types array became out of order then I guess that this system might be necessary